api/catalog/api/controllers/search_controller.py

```diff
@@ -510,10 +510,18 @@ def _get_result_and_page_count(
         return 0, 1
 
     result_count = response_obj.hits.total.value
-    page_count = int(result_count / page_size)
-    if page_count % page_size != 0:
-        page_count += 1
-    if len(results) < page_size and page_count == 0:
-        result_count = len(results)
+    page_count = ceil(result_count / page_size)
+
+    if len(results) < page_size:
+        if page_count == 1:
+            result_count = len(results)
+
+        # If we have fewer results than the requested page size and are
```

Comment:
I was afraid that this change will mean that we have to update the frontend code to account for the `page_count` changing between request, but we actually do update it on every request, so it's fine :)